lib/msf/core/exploit/check_module.rb

```diff
@@ -0,0 +1,78 @@
+# -*- coding: binary -*-
+
+#
+# This mixin implements an exploit's check method by invoking an aux module
+#
+# NOTE: The module's run_host/run method MUST return an Msf::Exploit::CheckCode
+#
+
+module Msf
+module Exploit::Remote::CheckModule
+
+  def initialize(info = {})
+    super
+
+    register_advanced_options([
+      OptString.new('CheckModule', [true, 'Module to check with'])
+    ])
+  end
+
+  # When this mixin is included, this method becomes the exploit's check method
+  #
+  # @return [Msf::Exploit::CheckCode] Whether or not the target is vulnerable
+  def check
+    # Instantiate the module
+    mod = framework.modules.create(check_module)
+
+    # Bail if we couldn't
+    unless mod
+      print_error("Could not instantiate #{check_module}")
+      return CheckCode::Unsupported
+    end
+
+    # Bail if run_host/run isn't defined
+    if mod.respond_to?(:run_host)
+      meth = :run_host
+    elsif mod.respond_to?(:run)
+      meth = :run
+    else
+      print_error("#{check_module} does not define a run_host/run method")
+      return CheckCode::Unsupported
+    end
+
+    # Add the exploit's targeting options to the module's datastore
+    %w[RHOSTS RHOST RPORT].each do |opt|
+      mod.datastore[opt] = datastore[opt].dup
+    end
+
+    # Bail if options don't validate
+    mod.options.validate(mod.datastore)
```

Comment:
This will `raise`, but note that the command dispatcher will catch `RHOST(S)` and `RPORT` validation errors before getting to this point.